### PR TITLE
Technical feedback ads

### PR DESCRIPTION
**Pull request by uplne**
Adding creative IDs of ads displayed on the page to the technical feedback email to be able to find possible ad issues faster.

I've also added ophan id for the future reference.

screen shot 2015-10-30 at 12 40 33

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 1]
```diff
@@ -78,6 +78,7 @@ define([
         rendered             = false,
         slots                = {},
         slotsToRefresh       = [],
+        slotIDs              = [],
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: Could the name of this imply the IDs of the actual slot HTML elements?

File: static/src/javascripts/projects/common/modules/onward/tech-feedback.js [hunk 2]
```diff
@@ -30,7 +34,11 @@ define([
                 var oldHref = link.attr('href');
                 var props = {
                     page: window.location,
-                    width: window.innerWidth
+                    width: window.innerWidth,
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/onward/tech-feedback.js]: So, you're just grabbing the creativeID properties? Could you not just do

    ads : _.map(dfp.getCreativeIDs(), 'creativeID'),

- or, if you don't want to work with string keys -

    ads: _.map(dfp.getCreativeIDs(), function (creativeData) {return creativeData.creativeId;}

Comment (uplne) [inline static/src/javascripts/projects/common/modules/onward/tech-feedback.js]: I can. I had more parameters in the object before but I've deleted it so yes, the map is better now.

File: onward/app/views/feedback.scala.html [hunk 3]
```diff
@@ -25,6 +25,7 @@ <h1 itemprop="headline" class="content__headline">Solve a technical issue</h1>
                                             <li>crashing</li>
                                             <li>reloading</li>
                                             <li>very slow site access</li>
+                                            <li>display ads problem</li>
```
Comment (jbreckmckye) [inline onward/app/views/feedback.scala.html]: I'm not sure if this is the right phrasing? I would say `issues with advertising`. I'm not sure if 'display ads' might be too specialised vocabulary for the average punter.

**Comment by jbreckmckye** [comment]
I'm not sure about the user text, but otherwise, :+1:

**Comment by ScottPainterGNM** [comment]
Issues with advertising sounds better but this is in the wrong place, all
the top set of links all go to the self-help page. It would need to be
linked to the email feedback part of the page I think.

Overall excited though!

On 30 October 2015 at 15:39, Jimmy Breck-McKye <notifications@github.com>
wrote:

> I'm not sure about the user text, but otherwise, [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/11025#issuecomment-152559284>.
>



-- 

Scott Painter

QA Analyst/theguardian.com

Guardian News & Media

-----

scott.painter@theguardian.com <alex.breuer@theguardian.com>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Comment by uplne** [comment]
@ScottPainterGNM so I will remove the text

**Comment by ScottPainterGNM** [comment]
Yeah I think it would be good to have text there to allow people to specify
but maybe a great 'report ads' button would be better in the long run.

Loving this though, scraping for ad info when people email into dotcom
feedback will be hugely useful!

On 30 October 2015 at 15:45, Steve Vadocz <notifications@github.com> wrote:

> @ScottPainterGNM <https://github.com/ScottPainterGNM> so I will remove
> the text
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/11025#issuecomment-152560830>.
>



-- 

Scott Painter

QA Analyst/theguardian.com

Guardian News & Media

-----

scott.painter@theguardian.com <alex.breuer@theguardian.com>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Pull request merged**